Source/Core/DiscIO/VolumeFileBlobReader.cpp

```diff
@@ -33,6 +33,13 @@ VolumeFileBlobReader::VolumeFileBlobReader(const Volume& volume, const Partition
 {
 }
 
+std::unique_ptr<BlobReader> VolumeFileBlobReader::CopyReader() const
+{
+  return std::unique_ptr<VolumeFileBlobReader>(
+      new VolumeFileBlobReader(std::move(*CreateVolume(m_volume.GetBlobReader().CopyReader())),
+                               m_partition, m_file_info->CopyFileInfo()));
```

Comment:
This is unsafe. The new VolumeFileBlobReader keeps a reference to the volume, but the volume gets destroyed once this line is done executing.

Also, the `std::move` seems meaningless.